py/desispec/io/raw.py

```diff
@@ -55,6 +56,24 @@ def read_raw(filename, camera, fibermapfile=None, **kwargs):
             log.error("Did not find header keyword EXPTIME in any HDU of {}".format(filename))
             raise KeyError("Did not find header keyword EXPTIME in any HDU of {}".format(filename))
 
+    #- Fix occasional missing or None NIGHT (e.g. on 20210105/70934)
+    #- update header as needed so the correction propagates downstream
+    try:
+        night = int(primary_header['NIGHT'])
+    except (KeyError, ValueError, TypeError):
```

Comment:
If this fails, I don't think the variable "night" gets defined using this try statement. 

Also, is there a motivation for using try rather than just checking that the key exists and value isn't None? I appreciate that the net result is equivalent, so I'm fine with this.

Reply:
If you don't use the variable night, I think your function call already has this logic, so it might be redundant as well. You could just call it for everything and it will return without assigning anything of the header keyword is valid.

Reply:
The point here is to test if the keyword "NIGHT" exists and whether it can be converted into an int, and if not replace it with a valid NIGHT.  Although "and if not None" logic would work for the problem last night (20210105), on 20190626 the keyword existed, wasn't None, but was a blank string instead.  Rather than try to individually test for each way it could be wrong in the past and future, I opted for try/except for whether you can succeed to hopefully be robust to new and novel ways in which that keyword could be wrong in the future...

You make a good point about `night` not being set if the try fails.  I didn't actually need that variable but the meaningful name implied that maybe you could rely upon it, so I renamed it to `tmp` instead.